fs/inode/fs_files.c

```diff
@@ -278,10 +316,28 @@ void files_releaselist(FAR struct filelist *list)
     }
 
   kmm_free(list->fl_files);
+}
 
-  /* Destroy the mutex */
+/****************************************************************************
+ * Name: files_get_filecount
+ *
+ * Description:
+ *   Given a file descriptor, return the corresponding instance of struct
+ *   file.
+ *
+ * Input Parameters:
+ *   fd    - The file descriptor
+ *   filep - The location to return the struct file instance
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned on
+ *   any failure.
+ *
+ ****************************************************************************/
 
-  nxmutex_destroy(&list->fl_lock);
+int files_get_filecount(FAR struct filelist *list)
+{
+  return list->fl_rows * CONFIG_NFILE_DESCRIPTORS_PER_BLOCK;
```

Comment:
what about protection `list->fl_rows` from race conditions? I know that `list->fl_rows` will rarely be changed, but won't this make any bugs even rarer and harder to detect? or is there no negative consequences from this ?

Reply:
`fl_rows` will not shrink dynamically, there will be no side effect if it continues to increase.